reference/extensions/python_requires.rst

```diff
@@ -1,4 +1,304 @@
+``python_requires`` as we did before and we'd need to tell Conan the base classes to use
+in the attribute ``python_requires_extend``. Here our recipe will inherit from the
+class ``MyBase``:
+
+
+.. code-block:: python
+    
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "0.1"
+        python_requires = "pyreq/0.1"
+        python_requires_extend = "pyreq.MyBase"
+
+
+The resulting inheritance is equivalent to declare our ``Pkg`` class as ``class Pkg(pyreq.MyBase, ConanFile)``.
+So creating the package we can see how the methods from the base class are reused:
+
+.. code-block:: bash
+
+    $ conan create .
+    ...
+    pkg/0.1: My cool source!
+    pkg/0.1: My cool build!
+    pkg/0.1: My cool package!
+    pkg/0.1: My cool package_info!
+    ...
+
+
+In general, base class attributes are not inherited, and should be avoided as much as possible.
+There are method alternatives to some of them like ``export()`` or ``set_version()``.
+For exceptional situations, see the ``init()`` method documentation for more information to extend inherited attributes.
+
+
+Reusing files
+-------------
+
+It is possible to access the files exported by a recipe that is used with ``python_requires``.
+We could have this recipe, together with a *myfile.txt* file containing the "Hello" text.
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class PyReq(ConanFile):
+        name = "pyreq"
+        version = "1.0"
+        package_type = "python-require"
+        exports = "*"
+
+
+.. code-block:: bash
+
+    $ echo "Hello" > myfile.txt
+    $ conan create .
+
+
+Now the python-require has been created, we can access its path (the place where *myfile.txt* is) with the
+``path`` attribute:
+
+.. code-block:: python
+
+    import os
+
+    from conan import ConanFile
+    from conan.tools.files import load
+
+    class Pkg(ConanFile):
+        python_requires = "pyreq/0.1"
+
+        def build(self):
+            pyreq_path = self.python_requires["pyreq"].path
+            myfile_path = os.path.join(pyreq_path, "myfile.txt")
+            content = load(self, myfile_path)  # content = "Hello"
+            self.output.info(content)
+            # we could also copy the file, instead of reading it
+
+
+Note that only ``exports`` work for this case, but not ``exports_sources``.
+
+
+Testing python-requires
+-----------------------
+
+It is possible to test with ``test_package`` a ``python_require``, by adding a ``test_package/conanfile.py``:
+
+.. code-block:: python
+    :caption: conanfile.py
+
+    from conan import ConanFile
+
+    def mynumber():
+        return 42
+
+    class PyReq(ConanFile):
+        name = "pyreq"
+        version = "1.0"
+        package_type = "python-require"
+
+
+.. code-block:: python
+    :caption: test_package/conanfile.py
+
+    from conan import ConanFile
+
+    class Tool(ConanFile):
```

Comment:
it probably needs an explicit `test_type` here (and what's the correct one for python requires?)

Reply:
test_type has been removed in 2.0, not used anymore

Reply:
but it's not automatically adding requirements in 2.0, right?

Reply:
No, it has to be explicit (except ``python-requires``)

Reply:
why except python requires?